### DISCUSSION TITLE
Why so secretive about the API?

### DISCUSSION THREAD

**Discussion by i-am-the-slime** [markedjs/marked#2527] [Q&A] [answered]
This is essentially one function. Yet, I can't find out what happens when I feed invalid markdown to `marked.parse`. 
Can somebody enlighten me?

**Comment by i-am-the-slime** [comment]
Okay I found the live editor. Looks like it spits the raw string back out.
Good!

**Comment by calculuschild** [comment, marked as answer]
Any string input to Marked.js is technically "valid". No matter what string you feed into `marked.parse`, you will get some HTML out. It just might not be the HTML you want (garbage in, garbage out). I.e., you could do `marked.parse("ajl*gA#**@(*()**gg**s*G@(AUG)J*")` and you would get some HTML (`<p>ajl<em>gA#**@(</em>()<strong>gg</strong>s<em>G@(AUG)J</em></p>
`) out. There's nothing secretive about it.

**Reply by i-am-the-slime** [reply]
Thank you!

**Reply by calculuschild** [reply]
Right, you won't get an error message or anything. It literally just converts what you give it. You won't see something like "oh, you forgot an asterisk here", but instead we assume that the input with a missing asterisk is what the user intended and translate it as-given.

**Reply by i-am-the-slime** [reply]
Yes, that is actually perfectly fine.
